subcommands/keys/rotate_root.go

```diff
@@ -31,15 +34,44 @@ func init() {
 	}
 	subcommands.RequireFactory(rotate)
 	rotate.Flags().BoolVarP(&doRootSync, "sync-prod", "", false, "Make sure production root.json is up-to-date and exit")
+	rotate.Flags().BoolVarP(&initialRotation, "initial", "", false, "Used for the first customer rotation. The command will download the initial root key")
 	cmd.AddCommand(rotate)
 }
 
 func doRotateRoot(cmd *cobra.Command, args []string) {
 	factory := viper.GetString("factory")
 	credsFile := args[0]
-	assertWritable(credsFile)
-	creds, err := GetOfflineCreds(credsFile)
-	subcommands.DieNotNil(err)
+
+	var creds OfflineCreds
+
+	if initialRotation {
+		// This lacks in efficiency (we wind up reading root.json from
+		// from server twice and create the creds.tgz file twice), but it makes
+		// up for that in clarity and ease of testing.
```

Comment:
We can have as readable code without fetching root.json twice if we move lines 76-77 inside the else clause (and declare the root variable above).

Reply:
this is why code comments are overrated :) thanks for the tip. force push fixed